vllm/config/utils.py

```diff
@@ -27,3 +38,103 @@ def config(cls: ConfigT) -> ConfigT:
     script, which is invoked during the pre-commit checks.
     """
     return cls
+
+
+def normalize_value(x):
+    """Return a stable, JSON-serializable canonical form for hashing.
+
+    Order: primitives, special types (Enum, callable, torch.dtype, Path), then
+    generic containers (Mapping/Set/Sequence) with recursion.
+    """
+    # Fast path
+    if x is None or isinstance(x, (bool, int, float, str)):
+        return x
+
+    # Enums by value
+    if isinstance(x, enum.Enum):
+        return x.value
```

Comment:
x.value is an int?
Should do normalize_value(x.value) instead (I'm not completely sure it's always an int)